### ISSUE TITLE
Configure TypeScript to target a modern ECMAScript standard

### ISSUE THREAD

**Issue by liammulh** [numberscope/frontscope#226]
Since we are using `BigInt` literals, we should set the target (see https://www.typescriptlang.org/tsconfig#target) to be at least ES2020.

In VSCode, I'm seeing errors like:

```
BigInt literals are not available when targeting lower than ES2020.ts(2737)
```

As of this writing, our TypeScript configuration file looks like [this](https://github.com/numberscope/frontscope/blob/9026a69b2a74205afaeef54ebf295bad10e1673a/tsconfig.json):

```json
{
    "files": [],
    "references": [
        {
            "path": "./tsconfig.vite-config.json"
        },
        {
            "path": "./tsconfig.app.json"
        },
        {
            "path": "./tsconfig.vitest.json"
        }
    ]
}
```

**Comment by liammulh** [comment]
@gwhitney, do you think this should be closed? We discussed this in our last weekly meeting, and I recall you thought we should try to keep the configuration minimal and abide by the "if it ain't broke, don't fix it" maxim.

**Comment by gwhitney** [comment]
Well, but the VSCode UI complaining is a way in which things **are** broken. So I have no problem leaving this issue in place and the target being updated, as long as the (manual) testing for the eventual PR includes running error-free in VSCode...

**Comment by gwhitney** [comment]
I also think that the target standard should be explicit somewhere in our tsconfig. As far as I can see, it is not, which I suspect means we are just getting the current tools' default, which may change unexpectedly, with unexpected consequences.

**Comment by gwhitney** [comment]
In ui2, there is a `"lib": ["ES2022"],` compiler option. Perhaps this should simply be moved up to the root tsconfig? I will make this a part of #420, which is about testing and typechecking anyway.

**Comment by gwhitney** [comment]
Resolved in ui2 per #420. Closing.
